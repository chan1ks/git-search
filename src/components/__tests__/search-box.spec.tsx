import React from 'react';

import renderer from 'react-test-renderer';
import SearchBox from '../form/search-box';

describe('SearchBox', () => {
  it('renders correctly', () => {
    const mockedChangeFn = jest.fn((event) => {
      console.log('Mocked change function');
    });

    const mockedSubmitFn = jest.fn((event) => {
      console.log('Mocked submit function');
    });

    const tree = renderer
      .create(
        <SearchBox
          handleChange={mockedChangeFn}
          handleSubmit={mockedSubmitFn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
