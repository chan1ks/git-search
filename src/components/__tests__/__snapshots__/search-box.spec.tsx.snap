// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <input
      className="form-control no-border-radius--right"
      onChange={[Function]}
      placeholder="Search a user"
      type="text"
    />
    <button
      className="btn btn-primary no-border-radius--left"
      type="submit"
    >
      SEARCH
    </button>
  </div>
</form>
`;
